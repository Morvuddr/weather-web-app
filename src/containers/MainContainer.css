.Main{
    width: 90%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-content: space-between;
    height: 500px;
    margin: auto;
}

@media (min-width: 700px) {
    .Main{
        flex-direction: row;
        height: 250px;
    }
}