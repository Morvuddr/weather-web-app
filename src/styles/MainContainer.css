.Main{
    width: 90%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-content: space-between;
    height: 500px;
    margin: auto;
}

@media (min-width: 700px) {
    .Main{
        flex-direction: row;
        height: 250px;
    }
}

.currentCity{
    display: flex;
    flex-direction: column;
    width: 45%;
}

.currentCity h1 {
    text-align: left;
}

.additionalInfo {
    display: flex;
    flex-direction: row;
    align-items: center;
    align-content: space-between;
    justify-content: space-between;
}
.additionalInfo h1{
    font-size: 60px;
    margin: 0;
}
.weatherImage{
    width: 90px;
    height: 90px;
}