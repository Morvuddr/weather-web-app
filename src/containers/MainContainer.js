import React from 'react';
import './MainContainer.css'
import { WeatherInfo } from '../components/WeatherInfo';
import { Loader } from '../components/Loader';
import CurrentCityComponent from '../components/CurrentCityComponent'

export default class MainContainer extends React.Component {

    render() {
        const { weather, isLoading } = this.props;
        if (isLoading) {
            return (
                <div className="Main">
                    <Loader />
                </div>
            );
        }



        return (
            <div className="Main">
                {isLoading
                    ? <Loader />
                    : (
                        <>
                            <CurrentCityComponent weather={weather} />
                            <WeatherInfo weather={weather} />
                        </>
                    )
                }
            </div>
        );
    }


}