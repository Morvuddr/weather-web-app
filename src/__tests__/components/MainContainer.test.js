import renderer from 'react-test-renderer';
import React from 'react';
import MainContainer from '../../containers/MainContainer';

describe('MainContainer component', () => {

    test('matches the snapshot when local weather is loaded', () => {
        const city = {
            isLoading: false,
            error: false,
            weather: {
                name: 'Moscow',
                img: 'abcd',
                temperature: 'temp',
                wind: 'wind',
                cloudiness: 'cloudiness',
                pressure: 'pressure',
                humidity: 'humidity',
                location: '[location]'
            }
        };

        const tree = renderer.create(
            <MainContainer weather={city.weather} isLoading={city.isLoading} error={city.error} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('matches the snapshot when local weather is loading', () => {
        const city = {
            isLoading: true,
            error: false,
            weather: {
                name: 'Moscow',
                img: 'abcd',
                temperature: 'temp',
                wind: 'wind',
                cloudiness: 'cloudiness',
                pressure: 'pressure',
                humidity: 'humidity',
                location: '[location]'
            }
        };

        const tree = renderer.create(
            <MainContainer weather={city.weather} isLoading={city.isLoading} error={city.error} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('matches the snapshot on error', () => {
        const city = {
            isLoading: false,
            error: true,
            weather: {
                name: 'Moscow',
                img: 'abcd',
                temperature: 'temp',
                wind: 'wind',
                cloudiness: 'cloudiness',
                pressure: 'pressure',
                humidity: 'humidity',
                location: '[location]'
            }
        };

        const tree = renderer.create(
            <MainContainer weather={city.weather} isLoading={city.isLoading} error={city.error} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});