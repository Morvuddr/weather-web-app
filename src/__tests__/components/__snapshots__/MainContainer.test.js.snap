// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer component matches the snapshot on error 1`] = `
<div
  className="Main"
>
  <div
    className="Error"
  >
    <h1>
      Не удалось загрузить данные
    </h1>
  </div>
</div>
`;

exports[`MainContainer component matches the snapshot when local weather is loaded 1`] = `
<div
  className="Main"
>
  <div
    className="currentCity"
  >
    <h1>
      Moscow
    </h1>
    <div
      className="additionalInfo"
    >
      <img
        alt="Иконка погоды"
        className="weatherImage"
        src="abcd"
      />
      <h1>
        temp
      </h1>
    </div>
  </div>
  <div
    className="WeatherInfo"
  >
    <div
      className="Info"
    >
      <h3>
        Ветер
      </h3>
      <h4>
        wind
      </h4>
    </div>
    <div
      className="Info"
    >
      <h3>
        Облачность
      </h3>
      <h4>
        cloudiness
      </h4>
    </div>
    <div
      className="Info"
    >
      <h3>
        Давление
      </h3>
      <h4>
        pressure
      </h4>
    </div>
    <div
      className="Info"
    >
      <h3>
        Влажность
      </h3>
      <h4>
        humidity
      </h4>
    </div>
    <div
      className="Info"
    >
      <h3>
        Координаты
      </h3>
      <h4>
        [location]
      </h4>
    </div>
  </div>
</div>
`;

exports[`MainContainer component matches the snapshot when local weather is loading 1`] = `
<div
  className="Main"
>
  <div
    className="Loader"
  >
    <h1>
      Подождите, данные загружаются
    </h1>
    <div
      className="Spinner"
    />
  </div>
</div>
`;
